src/index.js

```diff
@@ -6,46 +6,52 @@ const tags = tagsW3.map(tag => tag.replace(/[<|>]/g, ""));
 
 // Wait for the DOM to be fully loaded
 document.addEventListener("DOMContentLoaded", function () {
-    uploadTagsFromLocalStorage(); // list all tags in localStorage
-    updateProgressBar();
-    // Find the form element and add a submit event listener
-    const form = document.querySelector("form");
-    form.addEventListener("submit", function (event) {
-        // on submit
-        event.preventDefault(); // Prevent the default form submission
-        submitVal(); // call our submitVal function
-    });
 
-    function submitVal() {
-        let val = document.getElementById("input1").value;
-        // remove all white spaces and non-alphabetical characters except . ! - 
-        let regex = /[^a-zA-Z!-.]/g;
-        // change input to lower case and apply regex
-        let sanitizedVal = val.toLowerCase().replace(regex, "");
+  uploadTagsFromLocalStorage(); // list all tags in localStorage
+  updateProgressBar();
+  // Find the form element and add a submit event listener
+  const form = document.querySelector("form");
+  form.addEventListener("submit", function (event) {
+    // on submit
+    event.preventDefault(); // Prevent the default form submission
+    submitVal(); // call our submitVal function
+  });
 
-        let li = "<li>" + sanitizedVal + "</li>";
-        // check if the input value is in the tags array
-        let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
-        if (found) {
-            document.getElementById("list").insertAdjacentHTML("beforeend", li);
-            saveTagToLocalStorage(sanitizedVal); // Save tag to localStorage
-            updateProgressBar();
-        }
-        document.getElementById("input1").value = "";
+  function submitVal() {
+    let val = document.getElementById("input1").value;
+    // remove all white spaces and non-alphabetical characters except . ! - 
+    let regex = /[^a-zA-Z!-.]/g;
+    // change input to lower case and apply regex
+    let sanitizedVal = val.toLowerCase().replace(regex, "");
+    let li = "<li>" + sanitizedVal + "</li>";
+    // check if the input value is in the tags array
+    let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
+    // get the list of inputs already made
+    const getList = localStorage.getItem("myTags");
```

Comment:
When an object is saved to local storage, we use JSON.stringify because the localStorage API only allows for storage of strings. So when retrieving items from storage we want to use JSON.parse() to convert the string back into its original data structure (in our case, an array). This allows us to work with the stored data in the same way we would with the original data structure. 

It still works without the two JSON conversions, but we are prone to errors. Here for example, we use includes() on the string received from storage, but the method is more efficient on arrays. So all is needed is to wrap the right hand side of the expression on line 30 in JSAN.parse( ... ).

Reply:
True, I remember now. Since I had no problems with writing it this way (checking also the output with `console.log()` if I get the array) I thought I am on the safe side.